compiler/rustc_trait_selection/src/traits/engine.rs

```diff
@@ -38,7 +38,7 @@ impl<'tcx> TraitEngineExt<'tcx> for dyn TraitEngine<'tcx> {
 
     fn new_in_snapshot(tcx: TyCtxt<'tcx>) -> Box<Self> {
         if tcx.sess.opts.unstable_opts.chalk {
-            Box::new(ChalkFulfillmentContext::new())
+            Box::new(ChalkFulfillmentContext::new_in_snapshot())
```

Comment:
Is this needed for the changes here?

Reply:
yes, because `ObligationCtxt::new_in_snapshot` is used in `fn recompute_ambiguous_candidates` inside of a snapshot